engine/src/test/java/com/agiletec/aps/system/services/user/AuthenticationProviderManagerTest.java

```
@@ -198,55 +199,103 @@ public void authenticate_1() throws Exception {
+            throw e;
         }
     }
-    
+
+    @Test
+    public void loadUserByUsername_1() throws Exception {
+        UserDetails activeUser = this.createMockUser("admin", true, false, false);
+        when(this.userManager.getUser(Mockito.anyString())).thenReturn(activeUser);
+        when(this.authorizationManager.getUserAuthorizations("admin")).thenReturn(this.createMockAuthorization());
+        org.springframework.security.core.userdetails.UserDetails userDetails
+                = this.authenticationProviderManager.loadUserByUsername("admin");
+        Assert.assertNotNull(userDetails);
+        Assert.assertEquals("admin", userDetails.getUsername());
+        Assert.assertEquals(2, userDetails.getAuthorities().size());
+        Mockito.verify(userManager, Mockito.times(1)).updateLastAccess(Mockito.any(UserDetails.class));
+        Mockito.verify(userManager, Mockito.times(0)).getUser(Mockito.anyString(), Mockito.anyString());
+        Mockito.verify(userManager, Mockito.times(1)).getUser(Mockito.anyString());
+        Mockito.verify(authorizationManager, Mockito.times(1)).getUserAuthorizations(Mockito.anyString());
+        Mockito.verifyZeroInteractions(this.tokenManager);
+    }
+
+    @Test(expected = UsernameNotFoundException.class)
+    public void loadUserByUsername_2() throws Exception {
+        when(this.userManager.getUser(Mockito.anyString())).thenReturn(null);
+        try {
+            org.springframework.security.core.userdetails.UserDetails userDetails
+                    = this.authenticationProviderManager.loadUserByUsername("username");
+            Assert.fail();
+        } catch (UsernameNotFoundException e) {
+            Mockito.verifyZeroInteractions(this.authorizationManager);
+            Mockito.verifyZeroInteractions(this.tokenManager);
+            throw e;
+        }
+    }
+
+    @Test(expected = RuntimeException.class)
```

Is RuntimeException the specific expectation here? Seems to generic. Should be asserting against the UsernameNotFoundException. Generic RuntimeException check could mask a real error (though unlikely).

---

Fixed